crates/typst/src/text/mod.rs

```diff
@@ -463,6 +462,49 @@ pub struct TextElem {
     #[ghost]
     pub hyphenate: Hyphenate,
 
+    /// The "cost" of hyphenation when laying out text.
+    /// A higher cost means the layout engine will hyphenate less often.
+    /// Costs are specified as a ratio of the default cost,
+    /// so `50%` will make text layout twice as eager to hyphenate,
+    /// while `200%` (or 2) will make it half as eager.
+    ///
+    /// Hyphenation is generally avoided by placing the whole word on the next line,
+    /// so a higher runt cost can result in awkward justification spacing.
+    ///
+    /// The default cost is an acceptable balance,
+    /// but some may find that it hyphenates too eagerly,
+    /// breaking the flow of dense prose.
+    /// A cost of 6 (six times the normal cost) may work better for such contexts.
+    pub hyphenation_cost: Option<Ratio>,
+
+    /// The "cost" of runts (lines with a single word) when laying out text.
+    /// A higher cost means the layout engine will try harder to avoid runts.
+    /// Costs are specified as a ratio of the default cost,
+    /// so `50%` will make text layout try half as hard to avoid runts,
+    /// while `200%` (or 2) will make it try twice as hard.
+    ///
+    /// Runts are avoided by placing more or fewer words on previous lines,
+    /// so a higher runt cost can result in more awkward in justification spacing.
+    ///
+    /// The default cost is an acceptable balance,
+    /// but some may find that it produces too many runts,
+    /// which some style guides strongly discourage.
+    /// A cost of 6 (six times the normal cost) may work better for such contexts.
+    pub runt_cost: Option<Ratio>,
+
+    /// Whether to prevent widows (single line of paragraph on the next page)
+    /// and orphans (single line of paragraph on the previous page).
+    ///
+    /// Text layout prevents widows and orphans by default
+    /// because they are generally discouraged by style guides.
+    /// However, in some contexts they should not be avoided because
+    /// the prevention method, which moves a line to the next page,
+    /// can result in an uneven number of lines between pages,
+    /// which leads to asymmetry and breaking the flow of dense prose.
+    /// This option allows disabling any such modification.
+    #[default(true)]
+    pub prevent_widows_and_orphans: bool,
```

Comment:
`widow_and_orphan_prevention: bool` might be a more in line with the typical parameter naming, though I'm not sure on this one. It's also very long, but I don't see an obvious way to shorten it.

Reply:
I like that suggestion better and yeah I also failed to find a more concise name.